orangecontrib/spectroscopy/data.py

```diff
@@ -1014,16 +1014,195 @@ def read_v2(self):
+        print('html carregado')
+
+        # print(data_gsf_a, data_gsf_p, info)
+        # data = data_gsf_p
+        # return data, self.filename.split('.')[-1]
+        final_data, parameters, final_metas = self._format_file(data_gsf_a, data_gsf_p, info)
+        print('dado formatado')
+        # print(parameters)
+
+
+        metas = [Orange.data.ContinuousVariable.make("run"),
+                Orange.data.ContinuousVariable.make("row"),
+                Orange.data.ContinuousVariable.make("column"),
+                Orange.data.StringVariable.make("channel")]
+        print(metas)
+
+        domain = Orange.data.Domain([], None, metas=metas)
+        meta_data = Table.from_numpy(domain, X=np.zeros((len(final_data), 0)),
+                                     metas=np.asarray(final_metas, dtype=object))
+        
+        print(domain)
+        # meta_data.attributes = parameters
+
+        depth = np.arange(0, int(parameters['Pixel Area (X, Y, Z)'][3]))
+
+        print('meta_data', meta_data)
+        print('final_metas', final_metas)
+        print('data', final_data)
+        print('parametros', parameters)
+
+        # print(final_data)
+
+        return depth, final_data, meta_data
+
+    def _format_file(self, gsf_a, gsf_p, parameters):
+        
+        info = {}
+        for row in range(len(parameters)):
+            info.update({parameters[row,0].replace(':', ''): parameters[row,1:]})
+
+        print('escreveu o dicionario')
+
+        averaging = int(info['Averaging'][1])
+        px_x = int(info['Pixel Area (X, Y, Z)'][1]) # 10
+        px_y = int(info['Pixel Area (X, Y, Z)'][2]) # 10
+        px_z = number_of_points = int(info['Pixel Area (X, Y, Z)'][3]) # 1024
+        description = info['Description'][1]
+
+        print('Encontrou as informacoes')
+
+        data_complete = []
+        final_metas = []
+        for y in range(0, px_y):
+            print(y)
+            amplitude = gsf_a[y].reshape((1, px_x * px_z * averaging))[0]
+            phase = gsf_p[y].reshape((1, px_x * px_z * averaging))[0]
+            i = 0
+            f = i + px_z
+            for x in range(0, px_x):
+                for run in range(0, averaging):
+                    print(run)
+                    data_complete += [amplitude[i:f]]
+                    data_complete += [phase[i:f]]
+                    print('antes do final_metas')
+                    
+                    print(self.channel_a)
+                    print(self.channel_p)
+                    final_metas += [[run,x, y, self.channel_a]]
+                    final_metas += [[run,x, y, self.channel_p]]
+                    i = f
+                    f = i + px_z
+                    print('apos o final_metas')
+        print('formatou')
+        print('info', info)
+        return np.asarray(data_complete), info, final_metas
+
+    def _html_reader(self, path):
+        from pandas import read_html
+        try:
+            parameters = np.asarray(read_html(path)[0])
```

Comment:
If so, we should consider adding lxml as a dependency. If reading is done by lxml, why do we also need pandas? Do we need both?

Reply:
I read html using pandas, but when a call the function `read_html` it need `lxml` already installed. I think it's like dependency of pandas.

Reply:
So. I checked. Pandas does not install `lxml` automatically, so we need to remember it to add to our project's requirements.

Did you perhaps look if there are any other good ways to read this table without involving so big dependencies? I usually look for what is the simplest tool for the job. That keeps the final installation size small. It is fine to use pandas here if you decide it is the only solution that makes sense, but please look for alternatives.

Reply:
I have looked for an alternative `html` reader as  an array, but all these involving the same dependencies `lxml`, so kept the same opening package.